lib/iris/_data_manager.py

```diff
@@ -382,6 +411,26 @@ def dtype(self):
         return result
 
     @property
+    def fill_value(self):
+        return self._fill_value
+
+    @fill_value.setter
+    def fill_value(self, fill_value):
```

Comment:
Following on from my comments below, I think the way this functions should work is as follows: if `fill_value` is not `None`, it constructs a dummy numpy masked array with dtype `self.dtype` and a fill value of `fill_value`. Then the actual value stored in `self._fill_value` is retrieved from that array. That way we exactly mimic what numpy does, except in the case where the fill value is `None`. What do you think?